src/coreclr/src/vm/debughelp.cpp

```diff
@@ -1016,10 +1016,6 @@ void PrintDomainName(size_t ob)
 
 #include "gcdump.h"
 
-#include "../gcdump/i386/gcdumpx86.cpp"
```

Comment:
Yes. This was breaking build due to duplicate definitions, or something like that, on x86. 

I think linking from static library was more tolerant to such issues, since linker could just pick the first dupe, but linking with OBJECT libraries is closer to building from source and exposed things like this.